frontend/src/utils/utils.ts

```diff
@@ -28,6 +28,22 @@ export const round = (value: number | string, decimals = 2) => {
   });
 };
 
+/**
+ * @notice Rounds to appropriate human readable decimals for the token
+ * @dev This is very fragile and should be replace with something robust, i.e. something that walks
+ * value and rounds considering sig-figs or similar
+ */
+export const roundTokenAmount = (amount: BigNumberish, token: { decimals: number; address: string }): string => {
```

Comment:
Per our offline discussion, we might change this to something like this (untested) snippet:

```typescript
// Show two sig figs if number is <= 1, and show two decimals otherwise
export function formatAmount(amount: BigNumberish, decimals: number) {
  const formattedAmount = parseFloat(formatUnits(amount, decimals))
  // I think toLocaleString with maximumFractionDigits=2 should also work instead of toPrecision
  if (formattedAmount <= 1) return formattedAmount.toPrecision(2) 
  return formattedAmount.toLocaleString(undefined, {
      minimumFractionDigits: 0,
      maximumFractionDigits: 10,
    })
}
```

Would we still need the existing `round` and `formatAmount` methods used in various places, or could we use this everywhere?

Reply:
Agree with all of this. Mind if we pull into a separate issue and address it all at once?

Reply:
Do you mean leave this PR as-is and implement that method in the separate issue, or implement it now but pull out just the "use the new method everywhere else" part into a separate issue?

Reply:
The former. The feedback loop on testing this is rather slow so if there's no show stoppers on the current implementation, it'd be nice to get it in so we can launch WETH support soon, and clean it all up at once.

Reply:
Sounds good to me!

Reply:
Issue created [here](https://github.com/ScopeLift/umbra-protocol/issues/283)